rpc/namespaces/eth/api.go

```diff
@@ -300,17 +326,38 @@ func (api *PublicEthereumAPI) GetBlockTransactionCountByHash(hash common.Hash) *
 	return &n
 }
 
-// GetBlockTransactionCountByNumber returns the number of transactions in the block identified by number.
+// GetBlockTransactionCountByNumber returns the number of transactions in the block identified by its height.
 func (api *PublicEthereumAPI) GetBlockTransactionCountByNumber(blockNum rpctypes.BlockNumber) *hexutil.Uint {
 	api.logger.Debug("eth_getBlockTransactionCountByNumber", "block number", blockNum)
-	height := blockNum.Int64()
-	resBlock, err := api.clientCtx.Client.Block(&height)
+
+	var height int64
+	if blockNum != rpctypes.PendingBlockNumber {
+		if blockNum == rpctypes.LatestBlockNumber {
+			bn, err := api.backend.BlockNumber()
+			if err != nil {
+				return nil
+			}
+			bnCleaned := strings.Replace(bn.String(), "0x", "", -1)
+			height, err = strconv.ParseInt(bnCleaned, 16, 64)
+		} else {
+			height = blockNum.Int64()
+		}
+		resBlock, err := api.clientCtx.Client.Block(&height)
+		if err != nil {
+			return nil
+		}
+		txCount := hexutil.Uint(len(resBlock.Block.Txs))
+		return &txCount
+	}
+
+	// get the pending transaction count
+	pendingTxs, err := api.backend.PendingTransactions()
 	if err != nil {
 		return nil
 	}
 
-	n := hexutil.Uint(len(resBlock.Block.Txs))
-	return &n
+	txCount := hexutil.Uint(len(pendingTxs) / 2)
```

Comment:
why is this?

Reply:
strange behavior that i mentioned a month or 2 back but the pending queue is filled with `<nil>` for as many transactions are in the queue. So for example, if you send 2 transactions, the output of the pendingQueue will be `[<nil>, <nil>, &<0x123> $<0x456>]`
so the division is just a way to disregard the `nil`'s

Reply:
uh? we should fix that then